configfile_generator.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+from dd4hep import Detector
+import re
+from argparse import ArgumentParser
+import pprint
+import json
+
+parser = ArgumentParser()
+
+parser.add_argument(
+    "--compactFile", help="DD4hep compact description xml", required=True
+)
+parser.add_argument(
+    "--maxDepth",
+    help="Maximum traversal depth of the detector tree",
+    default=10,
+    type=int,
+)
+
+args = parser.parse_args()
+
+theDetector = Detector.getInstance()
+theDetector.fromXML(args.compactFile)
+start = theDetector.world()
+
+def process_name(raw_name):
+    name = re.sub(r"\d+", ".*", raw_name)
+    return name
+
+def tree(detElement, depth, maxDepth):
+    nd = {}
+    depth += 1
+    children = detElement.children()
+    for raw_name, child in children:
+        if depth > maxDepth:
+            tree(child, depth, maxDepth)
+        else:
+            dictionary = tree(child, depth, maxDepth)
+            nd.update({raw_name: dictionary})
+    return nd
+
+detector_dict = tree(start, 0, args.maxDepth)
+#pprint.pprint(detector_dict)
+
+def post_processing(obj, main_parts, subParts={}, sublist= []):
+    for k, v in obj.items():
+        if k in main_parts:
+            sublist = [f'{k}_(?!envelope)\\w+']
```

Comment:
For the people who don't know regex by heart (like me). Could you put a comment here briefly explaining what `(!?envelope)` does in the generated regular expression?